### ISSUE TITLE
Integrate as extension in piCorePlayer?

### ISSUE THREAD

**Issue by JWahle** [JWahle/piCoreCDSP#1]
Hi @paul-1,
I created this shell script to install CamillaDSP on piCorePlayer and was wondering, if you would like to integrate this as an extension into the piCorePlayer main repository.
Since I was not sure, what would be the right GitHub project to raise this issue, I created it in my project.

**Comment by paul-1** [comment]
The script overall looks good,  but I see cases where users would need to modify.

But perhaps a howto on docs.piCore player.org would be better suited for this.

**Comment by JWahle** [comment]
> But perhaps a howto on docs.piCore player.org would be better suited for this.

This would also be fine with me.

> The script overall looks good, but I see cases where users would need to modify.

Where do you see cases, where the users would need to modify?

Just for clarification, there are indeed a couple of steps, that do not go so well with the extension life-cycle and might benefit from some additional UI in piCorePlayer.
There is a little initial setup performed by the script:
- create the user config/filter directories with a default config in /mnt/mmcblk0p2/tce/camilladsp
- add the CDSP Alsa plugin to /etc/asound.conf
- set Squeezelite and ShairPort outputs to camilladsp in /usr/local/etc/pcp/pcp.cfg
- this is not implemented yet, but an initial config with the currently configured output device, that gives output in most cases could be generated

**Comment by paul-1** [comment]
The python stuff is pretty fluid.  For example, in the next pCP we will be using python3.11.

Couple of other thoughts.
-The 32bit can easily be auto detected.  'uname -m'
-Probably need to check for armv6 and say it's not supported.
-Are there fingerprints to validate the downloaded zip files?
-to update an extension, you do not need to remove the old one.  Just create /etc/sysconfig/tcedir/optional/upgrade and put the new extension in there, and it will be replaced during reboot.

**Comment by JWahle** [comment]
My general question is, if the piCore team would be interested in adding CamillaDSP to the piCorePlayer standard functionality, or the standard set of provided extensions.
If so, my script could serve as a basis for building the .tcz extension files - of course with modifications.
Currently, it is run on the user's piCorePlayer, but just because I didn't want to provide compiled files.
I believe, the piCorePlayer build is performed from some private repository, so I don't know how it would plug in there.

Thanks for sharing your suggestions, here is my reply to those:

> The python stuff is pretty fluid. For example, in the next pCP we will be using python3.11.

I guess, I might have to update my script, when new piCorePlayer versions are released - this was expected anyway.

> -The 32bit can easily be auto detected. 'uname -m'
> -Probably need to check for armv6 and say it's not supported.

Thanks for the suggestion, I just implemented detection of `aarch64` architecture.
Earlier, I decided against auto detection, because results of `uname -m` on 32bit systems seemed unpredictable, but that shouldn't be a problem, if I just detect for 64bit. Since I am not sure, how to exactly check for unsupported architectures, I will leave it out for now - unless you have further suggestions.

> -Are there fingerprints to validate the downloaded zip files?

Currently not. They would have to be added to the CamillaDSP and CamillaGUI repos, but GitHub currently doesn't help with that unfortunately.

> -to update an extension, you do not need to remove the old one. Just create /etc/sysconfig/tcedir/optional/upgrade and put the new extension in there, and it will be replaced during reboot.

I decided against supporting this use-case for this script, just to decrease the testing effort required by me, after script updates.
